src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/detentionengagementteam/DetentionEngagementTeamInternalAdaSuitabilityReviewPersonalisation.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.reform.iacasenotificationsapi.domain.personalisation.detentionengagementteam;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.AsylumCaseDefinition.*;
+
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.RequiredFieldMissingException;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.*;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.ccd.field.IdValue;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.personalisation.EmailWithLinkNotificationPersonalisation;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.service.DetEmailService;
+import uk.gov.hmcts.reform.iacasenotificationsapi.infrastructure.clients.DocumentDownloadClient;
+import uk.gov.service.notify.NotificationClientException;
+
+@Slf4j
+@Service
+public class DetentionEngagementTeamInternalAdaSuitabilityReviewPersonalisation implements EmailWithLinkNotificationPersonalisation {
+
+    private final String internalAdaSuitabilityReviewTemplateId;
+    private final DocumentDownloadClient documentDownloadClient;
+    private final DetEmailService detEmailService;
+
+    public DetentionEngagementTeamInternalAdaSuitabilityReviewPersonalisation(
+        @Value("${govnotify.template.adaSuitabilityReview.detentionEngagementTeam.internalAda.email}") String internalAdaSuitabilityReviewTemplateId,
+        DetEmailService detEmailService,
+        DocumentDownloadClient documentDownloadClient
+    ) {
+        this.internalAdaSuitabilityReviewTemplateId = internalAdaSuitabilityReviewTemplateId;
+        this.detEmailService = detEmailService;
+        this.documentDownloadClient = documentDownloadClient;
+    }
+
+    @Override
+    public String getTemplateId() {
+        return internalAdaSuitabilityReviewTemplateId;
+    }
+
+    @Override
+    public Set<String> getRecipientsList(AsylumCase asylumCase) {
+        Optional<String> detentionFacility = asylumCase.read(DETENTION_FACILITY, String.class);
+        if (detentionFacility.isEmpty() || !detentionFacility.get().equals("immigrationRemovalCentre")) {

```

Comment:
Better to have "immigrationRemovalCentre" as a private final (possibly static too) variable.